src/components/Header/Navbar.tsx

```diff
@@ -33,16 +36,20 @@ export const Navbar: React.FC = () => {
   const { projectIdentifier: projectFromUrl } = useParams<{
     projectIdentifier: string;
   }>();
+  const currProject = Cookies.get(CURRENT_PROJECT);
 
-  // Update current project cookie if the project in the URL does not equal the cookie value.
+  // Update current project cookie if the project in the URL is not an objectId and is not equal
+  // to the current project.
   // This will inform future navigations to the /commits page.
   useEffect(() => {
-    if (projectFromUrl && projectFromUrl !== Cookies.get(CURRENT_PROJECT)) {
+    if (
+      projectFromUrl &&
+      !validateObjectId(projectFromUrl) &&
```

Comment:
Great improvement! I think it's possible this has a large impact on the issue altogether.